src/Integration/Binding/DotNetBindingConfigProvider.cs

```diff
@@ -44,83 +45,98 @@ internal class DotNetBindingConfigProvider : IBindingConfigProvider
+            : this(sonarQubeService, nuGetBindingOperation, serverUrl, projectName, logger,
+                  new RuleSetGenerator(), new NuGetPackageInfoGenerator())
+        {
+        }
+
+        internal /* for testing */ DotNetBindingConfigProvider(ISonarQubeService sonarQubeService, INuGetBindingOperation nuGetBindingOperation, string serverUrl, string projectName, ILogger logger,
+            IRuleSetGenerator ruleSetGenerator, INuGetPackageInfoGenerator nuGetPackageInfoGenerator)
         {
             this.sonarQubeService = sonarQubeService;
             this.nuGetBindingOperation = nuGetBindingOperation;
             this.serverUrl = serverUrl;
             this.projectName = projectName;
             this.logger = logger;
+
+            this.ruleSetGenerator = ruleSetGenerator;
+            this.nuGetPackageInfoGenerator = nuGetPackageInfoGenerator;
         }
 
         public bool IsLanguageSupported(Language language)
         {
             return Language.CSharp.Equals(language) || Language.VBNET.Equals(language);
         }
 
-        public async Task<IBindingConfigFile> GetConfigurationAsync(SonarQubeQualityProfile qualityProfile, string organizationKey, Language language, CancellationToken cancellationToken)
+        public Task<IBindingConfigFile> GetConfigurationAsync(SonarQubeQualityProfile qualityProfile, string UNUSED_organizationKey_TODO_REMOVE, Language language, CancellationToken cancellationToken)
         {
-            if(!IsLanguageSupported(language))
+            if (!IsLanguageSupported(language))
             {
                 throw new ArgumentOutOfRangeException(nameof(language));
             }
 
+            return DoGetConfigurationAsync(qualityProfile, language, cancellationToken);
+        }
+
+        private async Task<IBindingConfigFile> DoGetConfigurationAsync(SonarQubeQualityProfile qualityProfile, Language language, CancellationToken cancellationToken)
+        {
             var serverLanguage = language.ServerLanguage;
             Debug.Assert(serverLanguage != null,
                 $"Server language should not be null for supported language: {language.Id}");
 
-            // Generate the rules configuration file for the language
-            var roslynProfileExporter = await WebServiceHelper.SafeServiceCallAsync(() =>
-                this.sonarQubeService.GetRoslynExportProfileAsync(qualityProfile.Name,
-                    organizationKey, serverLanguage, cancellationToken),
-                this.logger);
-            if (roslynProfileExporter == null)
-            {
-                this.logger.WriteLine(string.Format(Strings.SubTextPaddingFormat,
-                    string.Format(Strings.QualityProfileDownloadFailedMessageFormat, qualityProfile.Name,
-                        qualityProfile.Key, language.Name)));
-                return null;
-            }
+            // First, fetch the active rules
```

Comment:
We now fetch the data in multiple calls. I've optimised the calls so we don't fetch data until it is needed, so if the function returns early some of the data won't be fetch.